catlearn/graph_utils.py

```diff
@@ -536,3 +540,129 @@ def generate_random_graph(
     # return a randomly chosen graph in the factory's memory
     idx = random_generator.randint(0, factory.nb_graphs - 1)
     return factory.graphs[idx]
+
+
+def sample(
+        igraph: DirectedGraph[NodeType],
```

Comment:
There is only one graph around here, why not simply calling it "graph"?
If not I would prefer input_graph to igraph